os/kernel/task/task_create.c

```diff
@@ -128,6 +129,11 @@ static int thread_create(FAR const char *name, uint8_t ttype, int priority, int
 	pid_t pid;
 	int errcode;
 	int ret;
+	int tcb_size = sizeof(struct task_tcb_s);
+
+#ifndef CONFIG_DISABLE_PTHREAD
+	tcb_size = MAX(sizeof(struct task_tcb_s), sizeof(struct pthread_tcb_s));
```

Comment:
Please refer the commit message for this change.

I do understand that this function is not related to threads, but current implementation may produce very hard-to-catch bugs, and since the result might be used as `pthread_tcb_s` we have to ensure that there will be enough space for it.

Reply:
I don't understand what you want to say. As you agreed, this functions always uses task_tcb_s, never uses pthread_tcb_s. Why do you allocate maximun size between task and pthread?
If pthread_tcb_s is used inside that function, problem which we should fix is at another location. This is not a method of solving what you worried.